### ISSUE TITLE
NoMethodError: undefined method `win_friendly_path'

### ISSUE THREAD

**Issue by legal90** [chef-cookbooks/windows#553]
### Cookbook version
4.2.2

### Chef-client version
14.0.202

### Platform Details
Windows 2012R2

### Scenario:
Run list: `['omnibus::default']`, with omnibus cookbook v5.3.3. 

```
$ kitchen converge "default-windows-server-2012r2-standard"

<...cut...>

       Recipe: wix::default
         * remote_file[C:\Users\vagrant\AppData\Local\Temp\kitchen\cache/wix311-binaries.zip] action create
           - create new file C:\Users\vagrant\AppData\Local\Temp\kitchen\cache/wix311-binaries.zip
           - update content in file C:\Users\vagrant\AppData\Local\Temp\kitchen\cache/wix311-binaries.zip from none to da034c
           (file sizes exceed 10000000 bytes, diff output suppressed)
         * windows_zipfile[C:\wix] action unzip

           ================================================================================
           Error executing action `unzip` on resource 'windows_zipfile[C:\wix]'
           ================================================================================

           NoMethodError
           -------------
           undefined method `win_friendly_path' for #<#<Class:0x0000000005e707e8>:0x0000000003198220>

           Resource Declaration:
           ---------------------
           # In C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/wix/recipes/default.rb

            30: windows_zipfile node['wix']['home'] do
            31:   source download_path
            32:   overwrite true
            33:   action :nothing
            34: end
            35:
            36: # update path
            37: windows_path node['wix']['home'] do
            38:   action :add
            39: end

           Compiled Resource:
           ------------------
           # Declared in C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/cookbooks/wix/recipes/default.rb:30:in `from_file'

           windows_zipfile("C:\wix") do
             action [:nothing]
             default_guard_interpreter :default
             declared_type :windows_zipfile
             cookbook_name "wix"
             recipe_name "default"
             source "C:\\Users\\vagrant\\AppData\\Local\\Temp\\kitchen\\cache/wix311-binaries.zip"
             overwrite true
           end

           System Info:
           ------------
           chef_version=14.0.202
           platform=windows
           platform_version=6.3.9600
           ruby=ruby 2.5.1p57 (2018-03-29 revision 63029) [x64-mingw32]
           program_name=C:/opscode/chef/bin/chef-client
           executable=C:/opscode/chef/bin/chef-client


       Running handlers:
       [2018-05-02T14:03:06+02:00] ERROR: Running exception handlers
       [2018-05-02T14:03:06+02:00] ERROR: Running exception handlers
       Running handlers complete
       [2018-05-02T14:03:06+02:00] ERROR: Exception handlers complete
       [2018-05-02T14:03:06+02:00] ERROR: Exception handlers complete
       Chef Client failed. 15 resources updated in 01 minutes 22 seconds
       [2018-05-02T14:03:06+02:00] FATAL: Stacktrace dumped to C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/chef-stacktrace.out
       [2018-05-02T14:03:06+02:00] FATAL: Stacktrace dumped to C:/Users/vagrant/AppData/Local/Temp/kitchen/cache/chef-stacktrace.out
       [2018-05-02T14:03:06+02:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2018-05-02T14:03:06+02:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2018-05-02T14:03:06+02:00] FATAL: NoMethodError: windows_zipfile[C:\wix] (wix::default line 30) had an error: NoMethodError: undefined method `win_friendly_path' for #<#<Class:0x0000000005e707e8>:0x0000000003198220>
       [2018-05-02T14:03:06+02:00] FATAL: NoMethodError: windows_zipfile[C:\wix] (wix::default line 30) had an error: NoMethodError: undefined method `win_friendly_path' for #<#<Class:0x0000000005e707e8>:0x0000000003198220>
```

### Steps to Reproduce:
It's easy to reproduce with this TestKitchen suite:
https://github.com/chef-cookbooks/omnibus/blob/aa6c54da7f345c84dafe615077cd280665ea4424/.kitchen.yml#L48

The output is shown above. What is really weird is that it fails only once, on the initial run. If we run Chef once again, then it works fine.

### Expected Result:
The error should not be there.

### Actual Result:
See the output above

**Comment by legal90** [comment]
The issue was fixed by https://github.com/chef-cookbooks/windows/pull/555 🎉 
Thanks, @scotthain @tas50 !

Looking forward to get this fix released.

**Comment by tas50** [comment]
I shipped it as 4.2.3. Thanks for confirming the fix
